skimage/segmentation/morphsnakes.py

```diff
@@ -0,0 +1,440 @@
+                  lambda u: inf_sup(sup_inf(u))])  # ISoSI
+
+
+def _check_input(image, init_level_set):
+    """Check that shapes of `image` and `init_level_set` match."""
+    if len(image.shape) not in [2, 3]:
+        raise ValueError("Input image is not a 2D or 3D array.")
+
+    if len(image.shape) != len(init_level_set.shape):
+        raise ValueError("The dimensions of the initial level set do not "
+                         "match the dimensions of the image.")
+
+
+def _find_threshold(image, percentile=20):
+    """Find a proper threshold for `morph_gac` using the given percentile."""
+    return np.percentile(image, percentile)
+
+
+def _init_level_set(init_level_set, image_shape):
+    """Auxiliary function for initializing level sets with a string.
+
+    If `init_level_set` is not a string, it is returned as is.
+    """
+    if isinstance(init_level_set, str):
+        if init_level_set == 'checkerboard':
+            res = checkerboard_level_set(image_shape)
+        elif init_level_set == 'circle':
+            res = circle_level_set(image_shape)
+        else:
+            raise ValueError("`init_level_set` not in "
+                             "['checkerboard', 'circle']")
+    else:
+        res = init_level_set
+    return res
+
+
+def circle_level_set(image_shape, center=None, radius=None):
+    """Create a circle level set with binary values.
+
+    Parameters
+    ----------
+    image_shape : tuple of positive integers
+        Shape of the image
+    center : tuple of positive integers, optional
+        Coordinates of the center of the circle given in (row, column). If not
+        given, it defaults to the center of the image.
+    radius : float, optional
+        Radius of the circle. If not given, it is set to the 75% of the
+        smallest image dimension.
+
+    Returns
+    -------
+    out : array with shape `image_shape`
+        Binary level set of the circle with the given `radius` and `center`.
+
+    See also
+    --------
+    checkerboard_level_set
+    """
+
+    if center is None:
+        center = tuple(i // 2 for i in image_shape)
+
+    if radius is None:
+        radius = min(image_shape) * 3.0 / 8.0
+
+    grid = np.mgrid[[slice(i) for i in image_shape]]
+    grid = (grid.T - center).T
+    phi = radius - np.sqrt(np.sum((grid)**2, 0))
+    res = np.int8(phi > 0)
+    return res
+
+
+def checkerboard_level_set(image_shape, square_size=5):
+    """Create a checkerboard level set with binary values.
+
+    Parameters
+    ----------
+    image_shape : tuple of positive integers
+        Shape of the image.
+    square_size : int, optional
+        Size of the squares of the checkerboard. It defaults to 5.
+
+    Returns
+    -------
+    out : array with shape `image_shape`
+        Binary level set of the checkerboard.
+
+    See also
+    --------
+    circle_level_set
+    """
+
+    grid = np.mgrid[[slice(i) for i in image_shape]]
+    grid = (grid // square_size) & 1
```

Comment:
What does `& 1` do here?

Reply:
If it's converting to bool, please use `.astype(bool)`

Reply:
`& 1` is the way to make the square pattern of the checkerboard (it is like writing `% 2`). For example, in 1D, if `square_size` is 3, `grid // square_size` would be something like

`[0, 0, 0, 1, 1, 1, 2, 2, 2, 3, 3, 3...]`

and `(grid // square_size) & 1` would be

`[0, 0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 1...]`

which is a one-dimensional checkerboard. It should be integer, not `bool`.

Reply:
OooooOOOOOhhhh! Of course! Maybe change it to `% 2` and/or add a comment, "alternate 0/1 for even/odd numbers"?

Reply:
Following your suggestion, I added the comment and kept `& 1` (since `& 1` is several times faster than `% 2`). 

```python
In [6]: aux = np.random.randint(100, size=(1000000))

In [7]: %timeit aux & 1
1000 loops, best of 3: 907 µs per loop

In [8]: %timeit aux % 2
100 loops, best of 3: 18.1 ms per loop

In [9]: np.all((aux & 1) == (aux % 2))
Out[9]: True
```